### ISSUE TITLE
Add line hit testing and hover inspection along lines

### ISSUE THREAD

**Issue by bryevdv** [bokeh/bokeh#1486]
The major missing piece is a `_hit_line` method on the renderers that can return the indices of glyphs that are "hit" by a line. The primary use-cases would be horizontal and vertical lines (e.g., crosshair type inspector tools), and beginning with just these cases would be acceptable. 

Once line hit testing is in place, there simply needs to be a tool that requests a hit test against a line, and then pops up the hover when appropriate. There are a few possible paths here:

* add hit-testing as an option to existing crosshair tool 
* add line hit-testing as an option to existing hover tool
* create new tool specifically for line hit testing

**Comment by fpliger** [comment]
What about this?

![screen shot 2015-01-15 at 11 05 57](https://cloud.githubusercontent.com/assets/1084690/5755626/132a32de-9ca7-11e4-8814-18e95da62045.png)

**Comment by bryevdv** [comment]
Yes that is desired for sure, the main issue is the one about there being more than one kind of selection possible on a collection of lines. Do you want to know which *line*? Or which *points* in a line or lines? Either or both might be wanted in different circumstances. This possibility of "2D" selection indices is not accounted for at a conceptual level anywhere in the current plumbing so we just need to think about how we want to extend the current machinery.

**Comment by fpliger** [comment]
Yeah. I though that we can add this as hover behaviour on glyphs without _hit_point support. The downside is that it's an implicit behaviour... I think I prefer your proposal of having some options on the hover (maybe on inspect_tools in general?) that tells how to drive the tool behaviour...

**Comment by bryevdv** [comment]
We would need to implement `_hit_point` (or possibly `_hit_line`, thinking about a vertical crosshair case) to make any hover work in the first place, though.

**Comment by bryevdv** [comment]
status: low priority for 0.8
